Buttplug.Server.Managers.ErosTek/ET312Device.cs

```diff
@@ -0,0 +1,398 @@
+using System.Text;
+using System.Threading.Tasks;
+using System.Timers;
+using Buttplug.Core;
+using Buttplug.Core.Messages;
+
+namespace Buttplug.Server.Managers.ETSerialManager
+{
+    public class ET312Device : ButtplugDevice
+    {
+        private SerialPort serialPort;
+        private byte boxkey;
+        private double speed;
+        private double position;
+        private double currentPosition;
+        private double increment;
+        private double fade;
+        private double updateInterval;
+        private Timer updateTimer;
+        private object movementLock;
+
+        public ET312Device(SerialPort port, IButtplugLogManager aLogManager, string name, string id)
+            : base(aLogManager, name, id)
+        {
+            movementLock = new object();
+
+            // Handshake with the box
+            serialPort = port;
+            boxkey = 0;
+            Synch();
+
+            try
+            {
+                // Setup box for remote control
+                Execute(0x00);               // load default routine
+                Poke(0x4083, 0x00);          // disable front panel switches
+                Poke(0x409a, 0x00);          // Channel A: Gate Off
+                Poke(0x419a, 0x00);          // Channel B: Gate Off
+                Poke(0x40ac, 0x00);          // Channel A: Set intensity to static
+                Poke(0x41ac, 0x00);          // Channel B: Set intensity to static
+                Poke(0x40a5, 0x00);          // Channel A: Set intensity value
+                Poke(0x41a5, 0x00);          // Channel B: Set intensity value
+                Poke(0x409c, 0xff);          // stop volume ramp
+                Poke(0x40b5, 0x08);          // Channel A: MA knob sets frequency
+                Poke(0x41b5, 0x08);          // Channel B: MA knob sets frequency
+                Poke(0x40be, 0x04);          // Channel A: Set width from advanced menu
+                Poke(0x41be, 0x04);          // Channel B: Set width from advanced menu
+
+                // Let the user know we're in control now
+                WriteLCD("Buttplug", 8);
+                WriteLCD("----------------", 64);
+            }
+            catch
+            {
+                // If anything goes wrong during the setup
+                // consider the entire handshake failed
+                throw new ET312HandshakeException();
+            }
+
+            // We're now ready to receive events
+            MsgFuncs.Add(typeof(FleshlightLaunchFW12Cmd), HandleFleshlightLaunchFW12Cmd);
+            MsgFuncs.Add(typeof(StopDeviceCmd), HandleStopDeviceCmd);
+
+            // Start update timer
+            updateInterval = 20;                        // <- Change this value to adjust box update frequency in ms
+            updateTimer = new Timer();
+            updateTimer.Interval = updateInterval;
+            updateTimer.Elapsed += OnUpdate;
+            updateTimer.AutoReset = true;
+            updateTimer.Enabled = true;
```

Comment:
A neat C# feature is initialising object properties/fields "inline" (I can't remember what It's called, but you can change the above lines to:

```C#
// Start update timer
updateInterval = 20;                        // <- Change this value to adjust box update frequency in ms
updateTimer = new Timer(){
    Interval = updateInterval,
    AutoReset = true,
    Enabled,
};
updateTimer.Elapsed += OnUpdate;
```

It add to readability of code by easily identifying the setup of the `Timer` class